bin/_blackbox_common.sh

```diff
@@ -274,10 +284,37 @@ function enumerate_subdirs() {
     done
   done <"$listfile" | sort -u
 }
+ 
 
 # chdir to the base of the repo.
 function change_to_vcs_root() {
-  cd "$REPOBASE"
+  # if vcs_root not explicitly defined, use $REPOBASE
+  if [[ -z "$1" ]]; then
+    cd "$REPOBASE"
+
+  elif is_blackbox_repo "$1"; then
+    cd "$1"
+  
+  else
+    echo 'ERROR: $1 is not a blackbox Repo'
+    exit 1
+  fi
```

Comment:
Suggestion:  handle error conditions at top, with each exiting if needed. As a result, you'll need less indenting.

```
 # if vcs_root not explicitly defined, use $REPOBASE

 local rbase=${1:-$REPOBASE}  # Use $1 but if unset use $REPOBASE

 if ! is_blackbox_repo "$rbase"; then
   echo 'ERROR: $rbase is not a blackbox Repo'
   exit 1
 fi
 
 cd "$rbase"
```